src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -1227,6 +1226,8 @@ bool SotaUptaneClient::putManifestSimple() {
     storage->clearInstallationResults();
     return true;
   }
+
```

Comment:
```suggestion
At first glance it looks like the manifest assembling is redundant if there are pending updates, so I'd move line #1217 just after `if (hasPendingUpdates())` statement.
```

Reply:
Good point, but I'd need to double check. I have vague memories about this function having or used to having side-effects.